tests/rust/build.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import shlex
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dry-run", action="store_true")
+parser.add_argument("--verbose", action="store_true")
+parser.add_argument("--release", action="store_true")
+
+args = parser.parse_args()
+
+SYSTEMS = ['wasm32']
+VERSIONS = ['wasip1'] # + ['wasip2', 'wasip3']
+
+def compute_target(system, version):
+    return f"{system}-{version}"
+
+def compute_build_target(system, version):
+    if version == 'wasip3':
+        # wasm32-wasip3 triple not yet supported.
+        return compute_target(system, 'wasip2')
+    return compute_target(system, version)
+
+BASE_DIR = Path(__file__).parent
+
+def run(argv):
+    if args.verbose:
+        print(shlex.join([str(x) for x in argv]))
+    if not args.dry_run:
+        r = subprocess.run(argv)
+        if r.returncode != 0:
+            sys.exit(r.returncode)
```

Comment:
This should have similar effect
```suggestion
        r = subprocess.run(argv, check=True)
```

Reply:
Hmm, I kept the exit so as not to show the user a traceback, but I added a print for the command-line that could be re-run.